File: eBarbershop.Services/Database/PreporukaTermina.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eBarbershop.Services.Database
{
    public class PreporukaTermina
    {
        [Key]
        public int PreporukaId { get; set; }
        public int KlijentId { get; set; }
        public Korisnik Klijent { get; set; }
        public DateTime PreporuceniTermin { get; set; }
        public int KorisnikId { get; set; } // frizer
        public Korisnik Korisnik { get; set; }
        public int UslugaId { get; set; }
        public Usluga Usluga { get; set; }
        public double SkorPovjerenja { get; set; } // ML confidence score
        public string RazlogPreporuke { get; set; }
        public bool IsAccepted { get; set; }
    }
}

File: UI/ebarbershop_mobile/lib/models/preporuka_termina.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:ebarbershop_mobile/models/korisnik.dart';
import 'package:ebarbershop_mobile/models/usluga.dart';

part 'preporuka_termina.g.dart';

@JsonSerializable()
class PreporukaTermina {
  int? preporukaId;
  int? klijentId;
  Korisnik? klijent;
  DateTime? preporuceniTermin;
  int? korisnikId;
  Korisnik? korisnik;
  int? uslugaId;
  Usluga? usluga;
  double? skorPovjerenja;
  String? razlogPreporuke;
  bool? isAccepted;

  PreporukaTermina({
    this.preporukaId,
    this.klijentId,
    this.klijent,
    this.preporuceniTermin,
    this.korisnikId,
    this.korisnik,
    this.uslugaId,
    this.usluga,
    this.skorPovjerenja,
    this.razlogPreporuke,
    this.isAccepted,
  });

  factory PreporukaTermina.fromJson(Map<String, dynamic> json) => 
      _$PreporukaTerminaFromJson(json);
  
  Map<String, dynamic> toJson() => _$PreporukaTerminaToJson(this);
}